CodeEdit/Features/LSP/LanguageServer.swift

```diff
@@ -0,0 +1,193 @@
+//
+//  LanguageServer.swift
+//  CodeEdit
+//
+//  Created by Abe Malla on 2/7/24.
+//
+
+import JSONRPC
+import Foundation
+import LanguageClient
+import LanguageServerProtocol
+
+struct LanguageServer {
```

Comment:
`LanguageServer` seems like it would be a long living object, maybe make it a class?

Reply:
The LanguageServer struct only holds non-changing values and the functions that need them. They aren't ever copied around either or modified so no need for references. I kept them as a struct so they are more lightweight and because Swift recommends using structs by default and classes when you need class features, like inheritance. I believe the compiler can optimize better when we can throw out the possibilities of unused features.

Reply:
Only exception is the `trackedDocuments` variable which is intended to be removed in the next milestone as part of the new file solution.